packages/core/src/helpers.ts

```diff
@@ -25,12 +27,72 @@ export const formatDuration = (duration: number) => {
     }${milliseconds ? `${milliseconds}ms` : ''}`.trim();
 };
 
-export const getResolvedPath = (filepath: string) => {
-    try {
-        return require.resolve(filepath);
-    } catch (e) {
-        return filepath;
+// https://esbuild.github.io/api/#glob-style-entry-points
+const getAllEntryFiles = (filepath: string): string[] => {
+    if (!filepath.includes('*')) {
+        return [filepath];
     }
+
+    const files = glob.sync(filepath);
+    return files;
+};
+
+// Parse, resolve and return all the entries of esbuild.
+export const getEsbuildEntries = async (
+    build: PluginBuild,
+    context: GlobalContext,
+    log: Logger,
+): Promise<ResolvedEntry[]> => {
+    const entries: { name?: string; resolved: string; original: string }[] = [];
+    const entryPoints = build.initialOptions.entryPoints;
+    const entryPaths: { name?: string; path: string }[] = [];
+    const resolutionErrors: string[] = [];
+
+    if (Array.isArray(entryPoints)) {
+        for (const entry of entryPoints) {
+            const fullPath = entry && typeof entry === 'object' ? entry.in : entry;
+            entryPaths.push({ path: fullPath });
+        }
+    } else if (typeof entryPoints === 'object') {
+        entryPaths.push(
+            ...Object.entries(entryPoints).map(([name, filepath]) => ({ name, path: filepath })),
```

Comment:
Idk if this can happen, but if `entryPoints` is `null`, this will crash (typeof null is object)

Reply:
Good catch, it can happen (according to the types).